### PR TITLE
Fix mcsh5 channel order bug

### PR DESCRIPTION
**Pull request by alejoe91**
attempt to fix https://github.com/SpikeInterface/spiketoolkit/issues/275

@McD-NMI can you try this out and see if it fixes the problem and looks correct?

### CODE REVIEW

**Comment by McD-NMI** [comment]
Nope, does not work. Get a similar but different error: 

> ---------------------------------------------------------------------------
> TypeError                                 Traceback (most recent call last)
> c:\users\mcdonaldmatthew\pycharmprojects\organoidmea\venv\lib\site-packages\h5py\_hl\selections.py in select(shape, args, dsid)
>      84             try:
> ---> 85                 int(a)
>      86                 if isinstance(a, np.ndarray) and a.shape == (1,):
> 
> TypeError: only size-1 arrays can be converted to Python scalars
> 
> During handling of the above exception, another exception occurred:
> 
> ValueError                                Traceback (most recent call last)
> <ipython-input-10-a6ac5f50ab13> in <module>
>       3 
>       4 #filter signal
> ----> 5 recording_f = st.preprocessing.bandpass_filter(recording, freq_min=25, freq_max=6000, cache_to_file=True)
>       6 
>       7 #remove bad channels
> 
> c:\users\mcdonaldmatthew\pycharmprojects\organoidmea\venv\lib\site-packages\spiketoolkit\preprocessing\bandpass_filter.py in bandpass_filter(recording, freq_min, freq_max, freq_wid, type, order, chunk_size, cache_to_file, cache_chunks)
>     163     )
>     164     if cache_to_file:
> --> 165         return se.CacheRecordingExtractor(bpf_recording, chunk_size=chunk_size)
>     166     else:
>     167         return bpf_recording
> 
> c:\users\mcdonaldmatthew\pycharmprojects\organoidmea\venv\lib\site-packages\spikeextractors\cacherecordingextractor.py in __init__(self, recording, chunk_size, output_folder)
>      12         self._recording = recording
>      13         self._tmp_file = tempfile.NamedTemporaryFile(suffix=".dat", dir=output_folder).name
> ---> 14         dtype = recording.get_traces(start_frame=0, end_frame=2).dtype
>      15         recording.write_to_binary_dat_format(save_path=self._tmp_file, dtype=dtype, chunk_size=chunk_size)
>      16         BinDatRecordingExtractor.__init__(self, self._tmp_file, numchan=recording.get_num_channels(),
> 
> c:\users\mcdonaldmatthew\pycharmprojects\organoidmea\venv\lib\site-packages\spiketoolkit\preprocessing\filterrecording.py in get_traces(self, channel_ids, start_frame, end_frame)
>      39             ich1 = int(start_frame / self._chunk_size)
>      40             ich2 = int((end_frame - 1) / self._chunk_size)
> ---> 41             dt = self._recording.get_traces(start_frame=0, end_frame=1).dtype
>      42             filtered_chunk = np.zeros((len(channel_ids), (end_frame-start_frame)), dtype=dt)
>      43             pos = 0
> 
> c:\users\mcdonaldmatthew\pycharmprojects\organoidmea\venv\lib\site-packages\spikeextractors\extractors\mcsh5recordingextractor\mcsh5recordingextractor.py in get_traces(self, channel_ids, start_frame, end_frame)
>      92                 return recordings[sorted_idx] * conv
>      93             else:
> ---> 94                 return stream.get('ChannelData')[np.sort(channel_ids), start_frame:end_frame] * conv
>      95         else:
>      96             return stream.get('ChannelData')[np.array(channel_ids), start_frame:end_frame] * conv
> 
> h5py\_objects.pyx in h5py._objects.with_phil.wrapper()
> 
> h5py\_objects.pyx in h5py._objects.with_phil.wrapper()
> 
> c:\users\mcdonaldmatthew\pycharmprojects\organoidmea\venv\lib\site-packages\h5py\_hl\dataset.py in __getitem__(self, args)
>     551 
>     552         # Perform the dataspace selection.
> --> 553         selection = sel.select(self.shape, args, dsid=self.id)
>     554 
>     555         if selection.nselect == 0:
> 
> c:\users\mcdonaldmatthew\pycharmprojects\organoidmea\venv\lib\site-packages\h5py\_hl\selections.py in select(shape, args, dsid)
>      88             except Exception:
>      89                 sel = FancySelection(shape)
> ---> 90                 sel[args]
>      91                 return sel
>      92 
> 
> c:\users\mcdonaldmatthew\pycharmprojects\organoidmea\venv\lib\site-packages\h5py\_hl\selections.py in __getitem__(self, args)
>     399         self._id.select_none()
>     400         for idx, vector in enumerate(argvector):
> --> 401             start, count, step, scalar = _handle_simple(self.shape, vector)
>     402             self._id.select_hyperslab(start, count, step, op=h5s.SELECT_OR)
>     403 
> 
> c:\users\mcdonaldmatthew\pycharmprojects\organoidmea\venv\lib\site-packages\h5py\_hl\selections.py in _handle_simple(shape, args)
>     464         else:
>     465             try:
> --> 466                 x,y,z = _translate_int(int(arg), length)
>     467                 s = True
>     468             except TypeError:
> 
> c:\users\mcdonaldmatthew\pycharmprojects\organoidmea\venv\lib\site-packages\h5py\_hl\selections.py in _translate_int(exp, length)
>     484 
>     485     if not 0<=exp<length:
> --> 486         raise ValueError("Index (%s) out of range (0-%s)" % (exp, length-1))
>     487 
>     488     return exp, 1, 1
> 
> ValueError: Index (242) out of range (0-60)
>

**Comment by alejoe91** [comment]
@McD-NMI can you share wih me a mcsh5 file so I can test locally?

**Comment by alejoe91** [comment]
Also coud you just run the `recording.get_traces()` to see if that raises the same error?

**Comment by McD-NMI** [comment]
I tried some other things and noticed that when I call the recording.get_traces() function that it also fails. But if I specify the channel_ids it works: recording.get_traces(channel_ids=[242]) for example works. So I think the

channel_idxs = np.where(self._channel_ids == channel_ids)[0][0]

is working properly for this. The problem is when setting channel_idxs = self._channel_ids I will try some changes as well and share the .h5 file with you so you can test.

File: spikeextractors/extractors/mcsh5recordingextractor/mcsh5recordingextractor.py [hunk 1]
```diff
@@ -71,22 +71,30 @@ def get_traces(self, channel_ids=None, start_frame=None, end_frame=None):
         if end_frame is None:
             end_frame = self.get_num_frames()
         if channel_ids is None:
-            channel_idxs = self._channel_ids
+            channel_ids = self._channel_ids
```
Comment (samuelgarcia) [inline spikeextractors/extractors/mcsh5recordingextractor/mcsh5recordingextractor.py]: You said, no mix up between index and id before ?
:)

Comment (alejoe91) [reply spikeextractors/extractors/mcsh5recordingextractor/mcsh5recordingextractor.py]: It wa snot mixed up, just a different naming

**Comment by alejoe91** [comment]
> I tried some other things and noticed that when I call the recording.get_traces() function that it also fails. But if I specify the channel_ids it works: recording.get_traces(channel_ids=[242]) for example works. So I think the
> 
> channel_idxs = np.where(self._channel_ids == channel_ids)[0][0]
> 
> is working properly for this. The problem is when setting channel_idxs = self._channel_ids I will try some changes as well and share the .h5 file with you so you can test.

I think we should enforce np.array there. Can you try the latest commit?

**Comment by alejoe91** [comment]
@McD-NMI the last commit should fix it :) Can you try it?

**Comment by McD-NMI** [comment]
Yup, it seems to have fixed it. thanks so much!

**Comment by alejoe91** [comment]
> Yup, it seems to have fixed it. thanks so much!

No worries! Merging

**Pull request merged**